lib/yao/resources/user.rb

```diff
@@ -9,8 +9,11 @@ class User < Base
     self.resources_name = "users"
     self.admin          = true
 
-    def self.get_by_name(name)
-      self.list.find {|user| user.name == name }
+    class << self
+      def get_by_name(name)
+        self.get("", name: name)
```

Comment:
> GET /users?name=foo is user, not users

Wow. I understand.
I think `self.return_single_on_querying` flag is a nice idea. Please implement this way.